Libs/MRML/Widgets/qMRMLNodeComboBox.cxx

```diff
@@ -695,16 +708,44 @@ void qMRMLNodeComboBox::renameCurrentNode()
     return;
   }
 
+  // Use the dialog method to get the new name from the user
   bool ok = false;
-  QString newName = QInputDialog::getText(
-    this, "Rename " + this->nodeTypeLabel(node->GetClassName()), "New name:",
-    QLineEdit::Normal, node->GetName(), &ok);
-  if (!ok)
+  QString currentName = node->GetName();
+  QString newName = getNameFromDialog(renameTitlePrefix + this->nodeTypeLabel(node->GetClassName()), currentName, &ok);
+
+  // Check if the user confirmed the action
+  if (ok && !newName.isEmpty())
   {
-    return;
+    node->SetName(newName.toUtf8());
+    emit currentNodeRenamed(newName);
   }
-  node->SetName(newName.toUtf8());
-  emit currentNodeRenamed(newName);
+}
+// --------------------------------------------------------------------------
+void qMRMLNodeComboBox::createNodeAs(const QString& nodeTypeName)
+{
+    QString label = this->nodeTypeLabel(nodeTypeName);
+    bool okPressed = false;
+    QString dialogTitle = tr("Create New %1").arg(label);
```

Comment:
Shouldn't this use the already translated `createNew` from above?  Or do you think having the argument might mean it translates to a different string in different languages.